File: src/main/java/src/gestorAplicacion/personas/Profesor.java
package gestorAplicacion.personas;
import gestorAplicacion.Calendario.*;
import java.util.ArrayList;
import java.io.Serializable;

public class Profesor extends Persona{
	
	//atributos de instancia
	protected ArrayList<Materia> materias_Asignadas;
	private ArrayList<Double> calificacionesDocente;
	private double calificacionDocente;
	
	//constructor
	public Profesor (String nombre, int ID, String Email) {
		super(nombre, ID, Email);
		calificacionesDocente = new ArrayList<Double>();
		materias_Asignadas = new ArrayList<Materia>();
		calificacionDocente = 0;
	}
	//metodos set y get
	
	public void setCalificacionDocente(double calificacion) {
		this.calificacionDocente = calificacion;
	}
	
	public double getCalificacionDocente() {
		return calificacionDocente;
	}
	
	public void setCalificacionesDocente(ArrayList<Double> calificacionDocente) {
		this.calificacionesDocente=calificacionDocente;
	}
	
	public ArrayList<Double> getCalificacionesDocente() {
		return calificacionesDocente;
	}
	
	public ArrayList<Materia> getMaterias_Asignadas(){
		return materias_Asignadas;
	}
	
	public void setMaterias_Asignadas(ArrayList<Materia> materiasAsignadas) {
		this.materias_Asignadas= materiasAsignadas;
	}
	
	//metodos de clase
	
	public void asignarMateria(Materia nuevaMateria) {
		materias_Asignadas.add(nuevaMateria);
	}
	public void retirarMateria(Materia Materia) {
		materias_Asignadas.remove(Materia);
	}
	
	public void ingresarCalificacion(double calificacion) {
		calificacionesDocente.add(calificacion);
	}
	
	public void retirarCalificacion(double calificacion) {
		calificacionesDocente.remove(calificacion);
	}
	// Funcionalidad 4: Evaluacion docente
	public void evaluacionDocente() {
		double totalCalificaiones = 0;
		for(double calificacion: calificacionesDocente) {
			totalCalificaiones += calificacion;
		}
		calificacionDocente = Math.round((totalCalificaiones/calificacionesDocente.size())*10)/10.0;
	}
	
	public String toString() {
		return "Docente "+getNombre();
	}
	
}

File: src_py/personas/Profesor.py
from personas.Persona import Persona
from typing import List

class Profesor(Persona):
    def __init__(self, nombre: str, ID: int, Email: str):
        super().__init__(nombre, ID, Email)
        self.calificaciones_docente = []
        self.__materias_asignadas = []
        self.calificacion_docente = 0

    def setCalificacion_docente(self, calificacion):
        self.calificacion_docente = calificacion

    def getCalificacion_docente(self):
        return self.calificacion_docente

    def setMaterias_asignadas(self):
        return self.__materias_asignadas

    def getMaterias_asignadas(self, materias_asignadas):
        self.__materias_asignadas = materias_asignadas

    #Metodos de clase
    def asignar_materia(self, nueva_materia):
        self.__materias_asignadas.append(nueva_materia)

    def retirar_materia(self, materia):
        self.__materias_asignadas.remove(materia)

    def ingresar_calificacion(self, calificacion):
        self.calificaciones_docente.append(calificacion)

    def retirar_calificacion(self, calificacion):
        self.calificaciones_docente.remove(calificacion)

    def evaluacion_docente(self):
        total_calificaciones = 0
        for calificacion in self.calificaciones_docente:
            total_calificaciones += calificacion
        self.calificacion_docente = round((total_calificaciones/len(self.calificaciones_docente))*10)/10.0

    def __str__(self):
        return f'Docente {self.getNombre()}'
